packages/network-controller/src/create-auto-managed-network-client.ts

```diff
@@ -59,12 +62,15 @@ const UNINITIALIZED_TARGET = { __UNINITIALIZED__: true };
  *
  * @param networkClientConfiguration - The configuration object that will be
  * used to instantiate the network client when it is needed.
+ * @param customFeatureRpcApiMiddlewares - An array of middlewares to be added
+ * to the JSON RPC engine.
  * @returns The auto-managed network client.
  */
 export function createAutoManagedNetworkClient<
   Configuration extends NetworkClientConfiguration,
 >(
   networkClientConfiguration: Configuration,
+  customFeatureRpcApiMiddlewares: JsonRpcMiddleware<JsonRpcParams, Json>[],
```

Comment:
Could this be an optional parameter? It would make this PR non-breaking or close, and save a lot of duplicated lines of `, []` in the tests.

It would also align with `customFeatureRpcApiMiddlewares` being added as an optional property in `NetworkControllerOptions`.

Just a thought. Feel free to disregard if the intention is to draw attention to this parameter. 

```suggestion
  customFeatureRpcApiMiddlewares: JsonRpcMiddleware<JsonRpcParams, Json>[] = [],
```